frontend/packages/dev-console/src/utils/pipeline-filter-reducer.ts

```diff
@@ -1,27 +1,16 @@
 import * as _ from 'lodash';
 
-export const pipelineFilterReducer = (pipeline): string => {
-  if (
-    !pipeline ||
-    !pipeline.latestRun ||
-    !pipeline.latestRun.status ||
-    !pipeline.latestRun.status.succeededCondition
-  ) {
-    return '-';
-  }
-  return pipeline.latestRun.status.succeededCondition;
-};
-
 export const pipelineRunStatus = (pipelineRun): string => {
-  if (
-    !pipelineRun ||
-    !pipelineRun.status ||
-    !pipelineRun.status.conditions ||
-    pipelineRun.status.conditions.length === 0
-  ) {
-    return null;
+  const conditions = _.get(pipelineRun, ['status', 'conditions'], []);
+  const isCancelled = conditions.find((c) =>
+    ['PipelineRunCancelled', 'TaskRunCancelled'].some((cancel) => cancel === c.reason),
```

Comment:
I didn't notice this before, a PLR will show the reason for a fail being a TaskRun was cancelled? We don't cancel TaskRuns... that is weird. You sure this correct?

I don't see this in the CLI:
```
$ tkn pipelinerun describe mock-app-pipeline-o259xj
Name:           mock-app-pipeline-o259xj
Namespace:      andrew
Pipeline Ref:   mock-app-pipeline

Status
STARTED        DURATION    STATUS
1 minute ago   9 seconds   Failed(PipelineRunCancelled)

Message
PipelineRun "mock-app-pipeline-o259xj" was cancelled (TaskRun "mock-app-pipeline-o259xj-install-deps-k29ts" was cancelled)

Resources
No resources

Params
No params

Taskruns
NAME                                          TASK NAME      STARTED        DURATION   STATUS
mock-app-pipeline-o259xj-install-deps-k29ts   install-deps   1 minute ago   ---        Failed(TaskRunCancelled)
```

Status is `${status}(${reason})`
```
STARTED        DURATION    STATUS
1 minute ago   9 seconds   Failed(PipelineRunCancelled)
```

Reply:
This is because, the RunStatus function is usable over both PipelineRuns and TaskRuns... The status block for both are same for all statuses. Except for cancellation, where the reason differs.

Reply:
So you don't pass always a PipelineRun? Sometimes you pass TaskRuns?

Reply:
yes, the tooltip and the bars use the function for getting the status. Perhaps a more generic name would avoid the confusion.

Reply:
The function name is `pipelineRunStatus` and it accepts a `pipelineRun`. Please correct the name and I will log a refactor ticket so we can use types in this file (it's imported by the place the types are, so it's circular if we try to use the types).

Reply:
@andrewballantyne I completely agree with you. However, there needs a lot of refactoring in the status area. The Visualization uses different set of functions and the Bars and Tooltips uses another set. We would need all of these to point to the same function just the way we have for Actions. Can we address this as a separate concern?

Reply:
Agreed. There are fundamental issues with that file that predates this work. Although increased misuse happened with this work, it was already in place when this work was started. Since we need to correct a lot of things around Pipelines already, I'm okay with this being delayed to another PR.